tools/networkmonitor/networkmonitor.nim

```diff
@@ -195,6 +196,32 @@ proc crawlNetwork(node: WakuNode,
 
     await sleepAsync(crawlInterval)
 
+proc retrieveDynamicBootstrapNodes(dnsDiscovery: bool, dnsDiscoveryUrl: string, dnsDiscoveryNameServers: seq[ValidIpAddress]): Result[seq[RemotePeerInfo], string] =
+  if dnsDiscovery and dnsDiscoveryUrl != "":
+    # DNS discovery
+    debug "Discovering nodes using Waku DNS discovery", url=dnsDiscoveryUrl
+
+    var nameServers: seq[TransportAddress]
+    for ip in dnsDiscoveryNameServers:
+      nameServers.add(initTAddress(ip, Port(53))) # Assume all servers use port 53
+
+    let dnsResolver = DnsResolver.new(nameServers)
+
+    proc resolver(domain: string): Future[string] {.async, gcsafe.} =
+      trace "resolving", domain=domain
+      let resolved = await dnsResolver.resolveTxt(domain)
+      return resolved[0] # Use only first answer
```

Comment:
Is is possible for `resolveTxt` to return an empty seq?

Reply:
I don't know 🤷🏼 

I just copied the method from `wakunode2/app.nim`, as this app is importing a no longer public function.